backupccl: sanitize passphrase in BACKUP job description

adityamaru opened the pull request:
This change ensures that the passphrase used to encrypt a BACKUP
is redacted in the BACKUP job description.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52509)
<!-- Reviewable:end -->

adityamaru commented:
Split this out of #52434.

adityamaru committed:
backupccl: sanitize passphrase in BACKUP job description

This change ensures that the passphrase used to encrypt a BACKUP
is redacted in the BACKUP job description.

Release note: None

adityamaru commented:
TFTR!

bors r=dt
